packages/transformer/src/test/standalone/IModelExporter.test.ts

```diff
@@ -73,4 +73,69 @@ describe("IModelExporter", () => {
 
     sourceDb.close();
   });
+
+  describe("exportRelationships", () => {
+    it("should not export invalid relationships", async () => {
+      const sourceDbPath = IModelTransformerTestUtils.prepareOutputFile("IModelExporter", "InvalidRelationship.bim");
+      const sourceDb = SnapshotDb.createEmpty(sourceDbPath, { rootSubject: { name: "invalid-relationships" } });
+
+      const categoryId = SpatialCategory.insert(sourceDb, IModel.dictionaryId, "SpatialCategory", new SubCategoryAppearance());
+      const sourceModelId = PhysicalModel.insert(sourceDb, IModel.rootSubjectId, `PhysicalModel`);
+      const physicalObjectProps: PhysicalElementProps = {
+        classFullName: PhysicalObject.classFullName,
+        model: sourceModelId,
+        category: categoryId,
+        code: Code.createEmpty(),
+      };
+      const physicalObject1 = sourceDb.elements.insertElement(physicalObjectProps);
+      const physicalObject2 = sourceDb.elements.insertElement(physicalObjectProps);
+      const physicalObject3 = sourceDb.elements.insertElement(physicalObjectProps);
+      const physicalObject4 = sourceDb.elements.insertElement(physicalObjectProps);
+
+      const invalidRelationshipsProps: RelationshipProps[] = [
+        // target element will be deleted
+        {
+          classFullName: GraphicalElement3dRepresentsElement.classFullName,
+          targetId: physicalObject1,
+          sourceId: physicalObject2,
+        },
+        // target and source elements are invalid
+        {
+          classFullName: GraphicalElement3dRepresentsElement.classFullName,
+          targetId: "",
+          sourceId: "",
+        },
+        // only target element is invalid
+        {
+          classFullName: GraphicalElement3dRepresentsElement.classFullName,
+          targetId: "",
+          sourceId: physicalObject3,
+        },
+        // only source element is invalid
+        {
+          classFullName: GraphicalElement3dRepresentsElement.classFullName,
+          targetId: physicalObject4,
+          sourceId: "",
+        },
+      ];
+
+      invalidRelationshipsProps.forEach((props) => sourceDb.relationships.insertInstance(props));
+      sourceDb.withSqliteStatement(`DELETE FROM bis_Element WHERE Id = ${parseInt(physicalObject1, 16)}`, (stmt) => stmt.next());
```

Comment:
can you use `deleteElementTree` from `@itwin/core-backend`? This will leave entries in the `bis_GeometricElement3d` table and is not the right way to delete this.

Reply:
In this way, it deletes a relationship too (as it should be). So the issue, that connectors introduce, is not replicated in this way

Reply:
my bad, try `sourceDb.elements.deleteElement`